Grammar/python.gram

```diff
@@ -429,7 +429,14 @@ yield_expr[expr_ty]:
     | 'yield' a=[star_expressions] { _Py_Yield(a, EXTRA) }
 
 arguments[expr_ty] (memo):
-    | a=args [','] { a }
+    | a=args [','] &')' { a }
+    | incorrect_arguments [',']
+
+incorrect_arguments:
+    | args ',' '*' { raise_syntax_error(p, "iterable argument unpacking follows keyword argument unpacking") }
```

Comment:
This would probably disallow things like `f(x=1, *[], **{})`. A negatve lookahead for a star thereafter should do it though.

Also, I feel somewhat awkward to have `raise_syntax_error` return an `int` and then assign it to a variable of type `void *`.

Reply:
> This would probably disallow things like `f(x=1, *[], **{})`. A negatve lookahead for a star thereafter should do it though.

Yes, but the idea is that that alternative must have parsed previously by other rule with higher precedence so you could never reach that option. I was missing a lookahead to make this work (added in 11163f4):

```
❯ PYTHONPARSER=new ./python 
Python 3.9.0a5+ (heads/some_errors-dirty:312f116214, Apr  5 2020, 23:23:07) 
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> def f(*a, **k): ...
... 
>>> f(x=1, *[], **{})
```

Reply:
Whops, but there is still something wrong with assignment expressions as my latest commit didn't makes now `spam(a=1, (b := 2)` fail :(

Reply:
Shouldn't that fail though?

Reply:
Yeah, but it fails now with a different error

Reply:
Oh, okay.

Reply:
I think I managed to fix it, but I am a bit tired and I may have missed something

Reply:
One thing I found that's failing is `f(1, x=2, 3)`. The text is `SyntaxError: positional argument follows keyword argument unpacking`. I'm also having difficulties coming up with something that's universal.

Reply:
:( I think this may be indicating that we are missing some other shape for this piece of the grammar. Maybe we would need to insert the errors among the normal rules instead of deferring them to the end.....

Reply:
That's what I thought as well. I could work on that tomorrow. 2:30 is too late for this. :laughing: 

On another note, would it be okay to make `raise_syntax_error` return NULL, or is there a function somewhere that requires that it returns an `int`?

Reply:
> That's what I thought as well. I could work on that tomorrow. 2:30 is too late for this. laughing

Yeah, the beginning of this week is going to be a bit challenging for me so if you are able to keep iterating that would be great! It would be good to see if @gvanrossum has some insight on this as it seems that maybe some other refactoring of the rule could help here.

> On another note, would it be okay to make `raise_syntax_error` return NULL, or is there a function somewhere that requires that it returns an `int`?

Yeah, that's something to fix at the end. But there are multiple ways to fix that.

Reply:
> Yeah, the beginning of this week is going to be a bit challenging for me so if you are able to keep iterating that would be great! It would be good to see if @gvanrossum has some insight on this as it seems that maybe some other refactoring of the rule could help here.

I'll have plenty of time (not only this week, but probably for the next three weeks or so), so I'll certainly try my best.

> Yeah, that's something to fix at the end. But there are multiple ways to fix that.
 
:+1: